.github/workflows/release-artifacts.yml

```diff
@@ -33,6 +33,10 @@ jobs:
         with:
           fetch-depth: 0
 
+      - name: Exit if not on a release-* branch
+        if: startsWith(github.ref, 'release-') == false
+        run: exit -1
```

Comment:
Is this needed, since we have the `tags` key in the `on`?

Reply:
Right this check is supposed to be about the current branch, though it may not do that as it stands, let me check more

Reply:
Removed, as it is meaningless with a tag